### DISCUSSION TITLE
Allowing Python-style invocation of forward()

### DISCUSSION THREAD

**Discussion by NiklasGustafsson** [xamarin/TorchSharp#563] [Ideas]
We can, in limited situations, support Python-style invocation of modules, i.e. `module(t)` instead of `module.forward(t)` by introducing: 

```C#
public static implicit operator Func<Tensor, Tensor>(Module module)
{
    return (Tensor input) => module.forward(input);
}
```
It would only work in situations where the module is converted to a function reference, such as a call or variable assignment, not directly in an expression. 

In other words, this would work:

```C#
Func<Tensor,Tensor> rel = ReLU();
x = rel(t);
```
but not this:

```C#
var rel = ReLU();
x = rel(t);
```

Is this silly or useful?

**Comment by lostmsu** [comment]
IMHO, it would be better to overload some operator a la piping.
